packages/sd-jwt/src/SdJwtApi.ts

```diff
@@ -0,0 +1,88 @@
+import type { SdJwtCreateOptions, SdJwtPresentOptions, SdJwtReceiveOptions, SdJwtVerifyOptions } from './SdJwtOptions'
+import type { SdJwtVcVerificationResult } from './SdJwtService'
+import type { SdJwtRecord } from './repository'
+import type { Query } from '@aries-framework/core'
+
+import { AgentContext, injectable } from '@aries-framework/core'
+
+import { SdJwtService } from './SdJwtService'
+
+/**
+ * @public
+ */
+@injectable()
+export class SdJwtApi {
+  private agentContext: AgentContext
+  private sdJwtService: SdJwtService
+
+  public constructor(agentContext: AgentContext, sdJwtService: SdJwtService) {
+    this.agentContext = agentContext
+    this.sdJwtService = sdJwtService
+  }
+
+  public async create<Payload extends Record<string, unknown> = Record<string, unknown>>(
+    payload: Payload,
+    options: SdJwtCreateOptions<Payload>
+  ): Promise<{ sdJwtRecord: SdJwtRecord; compact: string }> {
+    return await this.sdJwtService.create<Payload>(this.agentContext, payload, options)
+  }
+
+  /**
+   *
+   * Validates and stores an sd-jwt from the perspective of an holder
+   *
+   */
+  public async receive(sdJwtCompact: string, options: SdJwtReceiveOptions): Promise<SdJwtRecord> {
+    return await this.sdJwtService.receive(this.agentContext, sdJwtCompact, options)
+  }
+
+  /**
+   *
+   * Create a compact presentation of the sd-jwt.
+   * This presentation can be send in- or out-of-band to the verifier.
+   *
+   * Within the `options` field, you can supply the indicies of the disclosures you would like to share with the verifier.
+   * Also, whether to include the holder key binding.
+   *
+   */
+  public async present(sdJwtRecord: SdJwtRecord, options: SdJwtPresentOptions): Promise<string> {
+    return await this.sdJwtService.present(this.agentContext, sdJwtRecord, options)
+  }
+
+  /**
+   *
+   * Verify an incoming sd-jwt. It will check whether everything is valid, but also returns parts of the validation.
+   *
+   * For example, you might still want to continue with a flow if not all the claims are included, but the signature is valid.
+   *
+   */
+  public async verify<
+    Header extends Record<string, unknown> = Record<string, unknown>,
+    Payload extends Record<string, unknown> = Record<string, unknown>
+  >(
+    sdJwtCompact: string,
+    options: SdJwtVerifyOptions
+  ): Promise<{ sdJwtRecord: SdJwtRecord<Header, Payload>; validation: SdJwtVcVerificationResult }> {
+    return await this.sdJwtService.verify<Header, Payload>(this.agentContext, sdJwtCompact, options)
+  }
+
+  public async getCredentialRecordById(id: string): Promise<SdJwtRecord> {
+    return await this.sdJwtService.getCredentialRecordById(this.agentContext, id)
+  }
+
+  public async getAllCredentialRecords(): Promise<Array<SdJwtRecord>> {
+    return await this.sdJwtService.getAllCredentialRecords(this.agentContext)
+  }
+
+  public async findCredentialRecordsByQuery(query: Query<SdJwtRecord>): Promise<Array<SdJwtRecord>> {
```

Comment:
Following the convention used in other module APIs (and services in general) we'd call it `findAllCredentialRecordsByQuery`, to make more explicit about the fact that we expect multiple records to be retrieved. Actually, in most modules we make naming simpler (e.g. `findAllByQuery`), but I see it's not the case here and `W3cCredentialsApi` (the API this module is based on I guess).

Of course if we change this here and leave `W3cCredentialsApi` as is, it would not be consistent. And if we change it in `W3cCredentialsApi` as well we'll introduce a breaking change. But I just wanted to point out this to bother you 😄

Reply:
Ah good point! I think we can change this in a separate PR which changes both the w3c and this one.

Reply:
I am actually not sure what the best name would be here. I think `findAllByQuery` is the best as it is already scoped under `sdJwt` within the `agent.modules` api.